File: client/Assets/Network/Room/Responses/ResponseJoinRoomExisting.cs
using UnityEngine;

public class ResponseJoinRoomExistingEventArgs : ExtendedEventArgs {
    public short Status { get; set; }
    public string Message { get; set; }
    public int PlayerId { get; set; }
    public string Username { get; set; }

    public ResponseJoinRoomExistingEventArgs() {
        Event_id = Constants.SMSG_JOIN_ROOM_EXISTING;
    }
}

public class ResponseJoinRoomExisting : BaseNetworkResponse {
    private int playerId;
    private string username;

    public ResponseJoinRoomExisting() {
        Response_id = Constants.SMSG_JOIN_ROOM_EXISTING;
    }
    
    protected override void ParseResponseData() {
        playerId = DataReader.ReadInt(DataStream);
        username = DataReader.ReadString(DataStream);
    }
    
    public override ExtendedEventArgs Process() {
        ResponseJoinRoomExistingEventArgs args = new ResponseJoinRoomExistingEventArgs();
        args.Status = status;
        args.Message = message;
        args.PlayerId = playerId;
        args.Username = username;
        return args;
    }
}

File: server/src/main/java/com/zzhgl/app/networking/response/room/ResponseJoinRoomExisting.java
package com.zzhgl.app.networking.response.room;

import com.zzhgl.app.metadata.Constants;
import com.zzhgl.app.networking.response.GameResponse;
import com.zzhgl.app.utility.GamePacket;

public class ResponseJoinRoomExisting extends GameResponse {
    private int playerId;
    private String username;

    public ResponseJoinRoomExisting(int playerId, String username) {
        responseCode = Constants.SMSG_JOIN_ROOM_EXISTING;
        this.playerId = playerId;
        this.username = username;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addShort16((short)0); // Success status
        packet.addString(""); // Empty message
        packet.addInt32(playerId);
        packet.addString(username);
        return packet.getBytes();
    }
}
